docs/en/dev_log/log_encryption.md

```diff
@@ -189,60 +189,115 @@ You can now build and test.
+:::
 
 ### Decrypt ULogs
 
-Before you can analyze your encrypted logs, you will need to decrypt them.
-There is a Python script that can be used to decrypt logs in `Tools/decrypt_ulog.py`.
+The [decrypt_logs.py](https://github.com/PX4/PX4-Autopilot/blob/main/Tools/log_encryption/decrypt_logs.py) script can be used to decrypt encrypted logs in `/logs/encrypted`, creating the unencrypted logs in `/logs/decrypted`.
+
+The script needs access to the private key corresponds to the public key used to encrypt the logs on the device.
+By default it uses the private key in `keys/private/private_key.pem`, but you can also pass the full path of a key as a command line argument.
 
-When decrypting a `.ulge` file the script takes 3 arguments:
+The expected folder structure showing the location of encrypted logs, decrypted logs and the default private key is shown below:
 
-1. The encrypted log file.
-2. An empty string `''`.
-3. The decryption key (the RSA2048 `.pem` private key which is used to unwrap the symmetric key).
+```sh
+PX4-Autopilot/
+│
+├── logs/ # Main directory for logs
+│ ├── encrypted/ # Stores encrypted logs (.ulge)
+│ │ ├── log-YYYY-MM-DD_HH-MM-SS_ID.ulge # Encrypted logs
+│ │
+│ ├── decrypted/
+│ │ ├── log-YYYY-MM-DD_HH-MM-SS_ID.ulg # Regular PX4 logs
+|
+├── keys/ # Main directory for keys
+  ├── private/ # Stores private keys
+    ├── private_key.pem # RSA private key (2048-bit)
+```
 
-For example:
+Run the tool as shown below (assuming the default key location):
 
 ```sh
-python3 decrypt_ulog.py \
-/home/john/Downloads/log_24_2024-10-6-23-39-50.ulg '' \
-new_keys/private_key.pem
+cd PX4-Autopilot/Tools/log_encryption
+python3 decrypt_logs.py
 ```
 
-On success the decrypted log file is created with the `.ulog` suffix.
+On success the decrypted logs can be found in the decrypted folder.
 
-::: info
-The script can be used with both `.ulge` logs and the `.ulgc`/`.ulgk` files used in [PX4 v1.15 Log Encryption](https://docs.px4.io/v1.15/en/dev_log/log_encryption.html).
-The full command line syntax is given below:
+## Generate RSA Public & Private Keys
 
-```sh
-usage: decrypt_ulog.py [-h] [ulog_file] [ulog_key] [rsa_key]
+The [Tools/log_encryption/generate_keys.py](https://github.com/PX4/PX4-Autopilot/blob/main/Tools/log_encryption/generate_keys.py) script can be used to generate the public key that is used on the device for encryption, and the private key that is used on the computer as part of log decryption.
 
-CLI tool to decrypt an ulog file
+The script is used as shown:
+
+```sh
+cd PX4-Autopilot/Tools/log_encryption
+python3 generate_keys.py
+```
 
-positional arguments:
-  ulog_file   .ulge/.ulgc, encrypted log file
-  ulog_key    .ulgk, legacy encrypted key (give empty string '' to ignore for .ulge)
-  rsa_key     .pem format key for decrypting the ulog key
+The private and public key will be generated as shown below.
+The script does not overwrite the existing keys and will also work with an existing private key.
```

Comment:
What does this mean? I mean isn't `private_key.pem` already present - so don't you have to overwrite it?

Reply:
Excellent. Have subedited to say that.